### DISCUSSION TITLE
Pax Web v8 and WebSockets

### DISCUSSION THREAD

**Discussion by j3rem1e** [ops4j/org.ops4j.pax.web#1946] [Q&A] [answered]
I'm migrating an application from Pax Web v7 to Pax Web v8, and I have an issue with my websocket (upgrading from Karaf 4.3 to Karaf 4.4). I don't use "war" but only the HTTP Whiteboard.

In Pax Web v7, my websocket was discovered through classpath scanning - due to some issues with the websockets whiteboard. In Pax Web v8, this scanning doesn't seem to work as before, but the whiteboard seems to be fixed. :)

Here are the updates I made:

1. My endpoint is now a real OSGi component.
2. I have added a "org.ops.pax.web.http.whiteboard.websocket=true" property.
3. I have installed the feature "pax-web-jetty-websockets".

After these updates, my websocket is discovered correctly and seems to be working correctly.

However, I was using a configurator to manage authentication with an HTTP Header, but this configurator is no longer being called.

Is there any documentation on this subject in Pax Web? How can I intercept the handshake now?

Thanks!

Jérémie

**Comment by grgrzybek** [comment, marked as answer]
There's an example of whiteboard registration here: https://github.com/ops4j/org.ops4j.pax.web/blob/web-8.0.27/pax-web-itest/pax-web-itest-container/pax-web-itest-container-common/src/main/java/org/ops4j/pax/web/itest/container/websockets/AbstractWebSocketWhiteBoardIntegrationTest.java#L69

And non-whiteboard registration is by installing a bundle here: https://github.com/ops4j/org.ops4j.pax.web/blob/web-8.0.27/pax-web-itest/pax-web-itest-container/pax-web-itest-container-common/src/main/java/org/ops4j/pax/web/itest/container/websockets/AbstractWebSocketIntegrationTest.java#L65-L67 where the bundle is a WAR with scanned component: https://github.com/ops4j/org.ops4j.pax.web/blob/web-8.0.27/samples/samples-war/war-websocket-jsr356/src/main/java/org/ops4j/pax/web/samples/websocket/ChatEndpoint.java#L30-L31

Because it all integrates within a _context_, you should be able to whiteboard-register your own `ServletContextHelper` with the authentication of your choice. Or simply add an authenticator to the WAR - I didn't check it and I see there are no websocket+auth samples in Pax Web (feel free to play with it ;).

**Reply by j3rem1e** [reply]
thanks!

Using the ServletContextHelper worked.
